data/mods/Magiclysm/items/spellbooks.json

```diff
@@ -606,7 +613,36 @@
     "looks_like": "cookbook",
     "symbol": "?",
     "color": "light_gray",
-    "use_action": { "type": "learn_spell", "spells": [ "taze", "laze", "technomancer_knifeshot" ] }
+    "use_action": {
+      "type": "learn_spell",
+      "spells": [ "taze", "laze", "technomancer_knifeshot", "technomancer_conjure_ups", "technomancer_recharge_vehicle" ]
+    }
+  },
+  {
+    "id": "techno_electrical_work",
+    "type": "BOOK",
+    "category": "manuals",
+    "name": { "str": "The Mage-Engineer's Practical Guide", "str_pl": "copies of The Mage-Engineer's Practical Guide" },
+    "//": "4 Technomancer spells, 2 Stormshaper spells",
+    "description": "This is a heavy, university thaumaturgical-program-level guidebook.  It discusses the various practical applications of magic in repairwork and engineering, with a focus on immediately-usable techniques instead of airy theoretical approaches.",
+    "weight": "4000 g",
+    "volume": "2500 ml",
+    "price": 5000,
+    "material": [ "paper" ],
```

Comment:
This breaks item density test:
```
Mods-(~[slow] ~[.],starting_items)=> ../tests/item_test.cpp:744: FAILED:
Mods-(~[slow] ~[.],starting_items)=>   CHECK( item_density( target ) <= max_density )
Mods-(~[slow] ~[.],starting_items)=> with expansion:
Error: Mods-(~[slow] ~[.],starting_items)=>   1.6f <= 1.2f
Mods-(~[slow] ~[.],starting_items)=> with messages:
Mods-(~[slow] ~[.],starting_items)=>   target.typeId() := string_id( "techno_electrical_work" )
Mods-(~[slow] ~[.],starting_items)=>   target.weight() := 4000000mg
Mods-(~[slow] ~[.],starting_items)=>   target.volume() := 2500ml
```

https://github.com/CleverRaven/Cataclysm-DDA/actions/runs/7702934299/job/20992679360?pr=71339#step:17:210

Reply:
Huh. Weird that the test here didn't catch it.

Reply:
The mapgen test failed so item density tests in Magicalysm was never run.